proposals/web-api-add-edit-with-media.md

```diff
@@ -0,0 +1,618 @@
+
+While OData streams works for the communication of the data, there are some media operations that cannot be preformed in real 
+time and are deferred post-processing that could prevent the successful publishing of the media.  This could include format
+validation, transcoding (resizing, reformating) for different displays/devices, compliance validation, copyright validation and/or content delivery distribution to name a few identified scenarios.  
+As a result of this need to allow for background operations to occur after the media has been received, a method to communicate the progression of the media is required.  The existing field `MediaStatus` will be used to comminucate the status and `MediaStatusDescription` will be introduced to provide contextual information when available.  This will assist clients in know why media has been rejected without having to contact the server provider outside of the transport.
+
+## Data Representation
+
+### Data Structure
+These are the required fields to represent the upload process for media.  The data dictionary change set can be found [here](https://docs.google.com/spreadsheets/d/1QGAe3hFvKHu9fMVzlGbI4gIuiSLjdTHO7leGK47bP88/edit?usp=sharing):
+
+```xml
+<EntityType Name="Media" hasStream="True">
+  ...
+  <Property Name="MediaStatus" Type="Edm.String" />
+  <Property Name="MediaStatusDescription" Type="Edm.String" />
+  ...
+</EntityType>
+```
+
+The `Media` resource/model will be annotated to have an OData stream attached.  
+
+### Status of the Media
+
+The status of the media will be tracked on the `Media` resource.  The `MediaStatus` field will show the status of the media.
+
+* `Incomplete` - the `Media` record that does not have a complete record 
+* `Processing` - the `Media` record is syntactically complete and is currently being processed by the backend
+* `Complete` - the `Media` record is validated and ready for consumption
+* `Deleted` - the `Media` record is no longer published 
+* `Rejected` - the `Media` has failed post processing for some reason.  Details of the problem will be presented in the `MediaStatusDescription` field.
+
+Some examples of `Rejected` status reason could be:
+* Media stream does not match `MediaType`  (Sent a video for a photo Media record)
+* The internal image checksum failed.
+* Image failed copyright ID check
+* Content of the media does not meet compliance rules
+
+## Media Add/Edit Process
+
+The creation of the media resource will be done using as a standard RESO Web API 2.1.0 Add/Edit process. The post SHOULD include all known resource record fields for the media with the exception of only the media byte stream.
+
+The response object MAY contain `mediaReadLink` and `mediaEditLink` annotations which the client MUST respect as per the OData specification when communicating the byte stream to the server. If fields are not provided the implicit URLs per the OData specification MUST then be used.  
```

Comment:
May want to provide a link to the section in the OData 4.01 specification that explains `mediaReadLink` and `mediaEditLink`.